src/reflect/type.go

```diff
@@ -673,13 +672,14 @@ func (t *rawType) rawField(n int) rawStructField {
 	// lookup faster), but by calculating it on-the-fly a bit of storage can be
 	// saved.
 	field := (*structField)(unsafe.Add(unsafe.Pointer(&descriptor.fields[0]), uintptr(n)*unsafe.Sizeof(structField{})))
-	offset := uintptr(field.offset)
 	data := field.data
 
 	// Read some flags of this field, like whether the field is an embedded
 	// field. See structFieldFlagAnonymous and similar flags.
 	flagsByte := *(*byte)(data)
 	data = unsafe.Add(data, 1)
+	offset, lenOffs := uvarint32(unsafe.Slice((*byte)(data), maxVarintLen32))
+	data = unsafe.Add(data, lenOffs)
```

Comment:
One thing to consider (although that might go a bit too far) is to not convert to a slice but work on raw pointers instead. That's a bit more dangerous, but reduces a little bit of binary size.